activiti-steps/activiti-steps-assertions-api/src/main/java/org/activiti/steps/matchers/TaskMatchers.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2019 Alfresco, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.activiti.steps.matchers;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.activiti.api.task.model.events.TaskRuntimeEvent;
+import org.activiti.api.task.runtime.events.TaskAssignedEvent;
+import org.activiti.api.task.runtime.events.TaskCompletedEvent;
+
+import static org.assertj.core.api.Assertions.*;
+
+public class TaskMatchers {
+
+    private TaskMatchers() {
+    }
+
+    public static TaskMatchers task() {
+        return new TaskMatchers();
+    }
+
+    public OperationScopeMatcher hasBeenAssigned() {
+        return (operationScope, events) -> {
+            List<TaskAssignedEvent> taskAssignedEvents = events
+                    .stream()
+                    .filter(event -> TaskRuntimeEvent.TaskEvents.TASK_ASSIGNED.equals(event.getEventType()))
+                    .map(TaskAssignedEvent.class::cast)
+                    .collect(Collectors.toList());
+            assertThat(taskAssignedEvents)
+                    .extracting(event -> event.getEntity().getId())
+                    .as("Unable to find event " + TaskRuntimeEvent.TaskEvents.TASK_ASSIGNED + " for task " + operationScope.getTaskId())
+                    .contains(operationScope.getTaskId());
+        };
+    }
+
+    public OperationScopeMatcher hasBeenCompleted() {
+        return (operationScope, events) -> {
+            List<TaskCompletedEvent> taskCompletedEvents = events
+                    .stream()
+                    .filter(event -> TaskRuntimeEvent.TaskEvents.TASK_COMPLETED.equals(event.getEventType()))
+                    .map(TaskCompletedEvent.class::cast)
+                    .collect(Collectors.toList());
+            assertThat(taskCompletedEvents)
+                    .extracting(event -> event.getEntity().getId())
+                    .as("Unable to find event " + TaskRuntimeEvent.TaskEvents.TASK_COMPLETED + " for task " + operationScope.getTaskId())
+                    .contains(operationScope.getTaskId());
+        };
+    }
+
+    public TaskResultMatcher hasAssignee(String assignee) {
```

Comment:
@erdemedeiros we need withCandidates() here..

Reply:
For now, I handled candidates with lambdas directly in the test using the library. In order to have it here, I think we'll need a CandidatesSource interface.

Reply:
👍